app/controllers/smiles_controller.rb

```diff
@@ -0,0 +1,13 @@
+class SmilesController < ApplicationController
+  def create
+    smile = Smile.where(user_id: current_user.id, photo_id: params[:photo_id]).first
+    if smile
+      smile.destroy
+    else
+      Smile.create(
+        user_id: current_user.id,
+        photo_id: params[:photo_id])
+    end
+    redirect_to '/photos'
+  end
```

Comment:
This should have two methods: `create` and `destroy`. Depending on which one you conditionally call from the HTML, you should change the target of the action.